### PR TITLE
Enhancement: 1. Added typescript-mode for ts[x] to autolist. 2: Added …

### PR DESCRIPTION
**Pull request by lanyusan**
…scala support based on scala metals. 3: Added three handy file manipuation functions to utils

### CODE REVIEW

File: lisp/init-utils.el [hunk 1]
```diff
@@ -34,6 +34,50 @@
   (require 'init-const)
   (require 'init-custom))
 
+
+;; Handy file manipulation functions
```
Comment (seagle0128) [inline lisp/init-utils.el]: Can you please split this to another PR?
These can be put in `init-funcs.el`.

Comment (lanyusan) [reply lisp/init-utils.el]: Why don't you accept it then do the move? It might be faster. :)

Comment (seagle0128) [reply lisp/init-utils.el]: I don't think it's a good idea submit several features in one PR. It's hard to manage especially when it breaks the build.
